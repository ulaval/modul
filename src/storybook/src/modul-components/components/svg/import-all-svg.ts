
import MSvgAddCircleFilled from '@ulaval/modul-components/dist/assets/icons/svg/add-circle-filled.svg';
import MSvgAddCircle from '@ulaval/modul-components/dist/assets/icons/svg/add-circle.svg';
import MSvgArrowDown from '@ulaval/modul-components/dist/assets/icons/svg/arrow--down.svg';
import MSvgArrowLeft from '@ulaval/modul-components/dist/assets/icons/svg/arrow--left.svg';
import MSvgArrowRight from '@ulaval/modul-components/dist/assets/icons/svg/arrow--right.svg';
import MSvgArrowUp from '@ulaval/modul-components/dist/assets/icons/svg/arrow--up.svg';
import MSvgArrowButtonDown from '@ulaval/modul-components/dist/assets/icons/svg/arrow-button--down.svg';
import MSvgArrowButtonLeft from '@ulaval/modul-components/dist/assets/icons/svg/arrow-button--left.svg';
import MSvgArrowButtonRight from '@ulaval/modul-components/dist/assets/icons/svg/arrow-button--right.svg';
import MSvgArrowButtonUp from '@ulaval/modul-components/dist/assets/icons/svg/arrow-button--up.svg';
import MSvgClock from '@ulaval/modul-components/dist/assets/icons/svg/clock.svg';
import MSvgLock from '@ulaval/modul-components/dist/assets/icons/svg/lock.svg';
import MSvgProfile from '@ulaval/modul-components/dist/assets/icons/svg/profile.svg';
import { ModulIconName } from '@ulaval/modul-components/dist/utils/modul-icons/modul-icons';
import { SvgSpriteService } from '@ulaval/modul-components/dist/utils/svg/svg-sprite';
import Vue from 'vue';

export const importAllSvg: () => void = () => {
    const svg: SvgSpriteService = Vue.prototype.$svgSprite;
    svg.addSvg(ModulIconName.AddCircleFilled, MSvgAddCircleFilled);
    svg.addSvg(ModulIconName.AddCircle, MSvgAddCircle);
    svg.addSvg(ModulIconName.ArrowDown, MSvgArrowDown);
    svg.addSvg(ModulIconName.ArrowLeft, MSvgArrowLeft);
    svg.addSvg(ModulIconName.ArrowRight, MSvgArrowRight);
    svg.addSvg(ModulIconName.ArrowUp, MSvgArrowUp);
    svg.addSvg(ModulIconName.ArrowButtonDown, MSvgArrowButtonDown);
    svg.addSvg(ModulIconName.ArrowButtonRight, MSvgArrowButtonRight);
    svg.addSvg(ModulIconName.ArrowButtonLeft, MSvgArrowButtonLeft);
    svg.addSvg(ModulIconName.ArrowButtonUp, MSvgArrowButtonUp);
    svg.addSvg(ModulIconName.Clock, MSvgClock);
    svg.addSvg(ModulIconName.Lock, MSvgLock);
    svg.addSvg(ModulIconName.Profile, MSvgProfile);
};
