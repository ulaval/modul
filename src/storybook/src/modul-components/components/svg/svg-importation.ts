
import MSvgAddCircleFilled from '@ulaval/modul-components/dist/assets/icons/svg/add-circle-filled.svg';
import MSvgAddCircle from '@ulaval/modul-components/dist/assets/icons/svg/add-circle.svg';
import MSvgArrowDown from '@ulaval/modul-components/dist/assets/icons/svg/arrow--down.svg';
import MSvgArrowLeft from '@ulaval/modul-components/dist/assets/icons/svg/arrow--left.svg';
import MSvgArrowRight from '@ulaval/modul-components/dist/assets/icons/svg/arrow--right.svg';
import MSvgArrowUp from '@ulaval/modul-components/dist/assets/icons/svg/arrow--up.svg';
import MSvgArrowButtonDown from '@ulaval/modul-components/dist/assets/icons/svg/arrow-button--down.svg';
import MSvgArrowButtonLeft from '@ulaval/modul-components/dist/assets/icons/svg/arrow-button--left.svg';
import MSvgArrowButtonRight from '@ulaval/modul-components/dist/assets/icons/svg/arrow-button--right.svg';
import MSvgArrowButtonUp from '@ulaval/modul-components/dist/assets/icons/svg/arrow-button--up.svg';
import MSvgArrowHeadCircleRight from '@ulaval/modul-components/dist/assets/icons/svg/arrow-head-circle--right.svg';
import MSvgArrowHeadFilledDown from '@ulaval/modul-components/dist/assets/icons/svg/arrow-head-filled--down.svg';
import MSvgArrowHeadFilledLeft from '@ulaval/modul-components/dist/assets/icons/svg/arrow-head-filled--left.svg';
import MSvgArrowHeadFilledRight from '@ulaval/modul-components/dist/assets/icons/svg/arrow-head-filled--right.svg';
import MSvgArrowHeadFilledUp from '@ulaval/modul-components/dist/assets/icons/svg/arrow-head-filled--up.svg';
import MSvgArrowReturn from '@ulaval/modul-components/dist/assets/icons/svg/arrow-return.svg';
import MSvgArrowThinDown from '@ulaval/modul-components/dist/assets/icons/svg/arrow-thin--down.svg';
import MSvgArrowThinLeft from '@ulaval/modul-components/dist/assets/icons/svg/arrow-thin--left.svg';
import MSvgArrowThinRight from '@ulaval/modul-components/dist/assets/icons/svg/arrow-thin--right.svg';
import MSvgArrowThinUp from '@ulaval/modul-components/dist/assets/icons/svg/arrow-thin--up.svg';
import MSvgAttachment from '@ulaval/modul-components/dist/assets/icons/svg/attachment.svg';
import MSvgAvatar from '@ulaval/modul-components/dist/assets/icons/svg/avatar.svg';
import MSvgCalendar from '@ulaval/modul-components/dist/assets/icons/svg/calendar.svg';
import MSvgChevronDown from '@ulaval/modul-components/dist/assets/icons/svg/chevron--down.svg';
import MSvgChevronLeft from '@ulaval/modul-components/dist/assets/icons/svg/chevron--left.svg';
import MSvgChevronRight from '@ulaval/modul-components/dist/assets/icons/svg/chevron--right.svg';
import MSvgChevronUp from '@ulaval/modul-components/dist/assets/icons/svg/chevron--up.svg';
import MSvgClock from '@ulaval/modul-components/dist/assets/icons/svg/clock.svg';
import MSvgCloseClear from '@ulaval/modul-components/dist/assets/icons/svg/close-clear.svg';
import MSvgCodeTag from '@ulaval/modul-components/dist/assets/icons/svg/code-tag.svg';
import MSvgCompletedFilled from '@ulaval/modul-components/dist/assets/icons/svg/completed-filled.svg';
import MSvgConfirmationWhiteFilled from '@ulaval/modul-components/dist/assets/icons/svg/confirmation-white-filled.svg';
import MSvgConfirmation from '@ulaval/modul-components/dist/assets/icons/svg/confirmation.svg';
import MSvgDelete from '@ulaval/modul-components/dist/assets/icons/svg/delete.svg';
import MSvgDownload from '@ulaval/modul-components/dist/assets/icons/svg/download.svg';
import MSvgDragVertical from '@ulaval/modul-components/dist/assets/icons/svg/drag-vertical.svg';
import MSvgDragndrop from '@ulaval/modul-components/dist/assets/icons/svg/dragndrop.svg';
import MSvgEdit from '@ulaval/modul-components/dist/assets/icons/svg/edit.svg';
import MSvgEmailSend from '@ulaval/modul-components/dist/assets/icons/svg/email-send.svg';
import MSvgEmail from '@ulaval/modul-components/dist/assets/icons/svg/email.svg';
import MSvgErrorFilled from '@ulaval/modul-components/dist/assets/icons/svg/error-filled.svg';
import MSvgErrorWhiteFilled from '@ulaval/modul-components/dist/assets/icons/svg/error-white-filled.svg';
import MSvgError from '@ulaval/modul-components/dist/assets/icons/svg/error.svg';
import MSvgEvaluation from '@ulaval/modul-components/dist/assets/icons/svg/evaluation.svg';
import MSvgFileAccess from '@ulaval/modul-components/dist/assets/icons/svg/file-access.svg';
import MSvgFileAudio from '@ulaval/modul-components/dist/assets/icons/svg/file-audio.svg';
import MSvgFileCode from '@ulaval/modul-components/dist/assets/icons/svg/file-code.svg';
import MSvgFileDefault from '@ulaval/modul-components/dist/assets/icons/svg/file-default.svg';
import MSvgFileDwg from '@ulaval/modul-components/dist/assets/icons/svg/file-dwg.svg';
import MSvgFileExcel from '@ulaval/modul-components/dist/assets/icons/svg/file-excel.svg';
import MSvgFileFlash from '@ulaval/modul-components/dist/assets/icons/svg/file-flash.svg';
import MSvgFileImage from '@ulaval/modul-components/dist/assets/icons/svg/file-image.svg';
import MSvgFileMarkup from '@ulaval/modul-components/dist/assets/icons/svg/file-markup.svg';
import MSvgFileMediaplayer from '@ulaval/modul-components/dist/assets/icons/svg/file-mediaplayer.svg';
import MSvgFileOpenofficeBase from '@ulaval/modul-components/dist/assets/icons/svg/file-openoffice-base.svg';
import MSvgFileOpenofficeCalc from '@ulaval/modul-components/dist/assets/icons/svg/file-openoffice-calc.svg';
import MSvgFileOpenofficeDefault from '@ulaval/modul-components/dist/assets/icons/svg/file-openoffice-default.svg';
import MSvgFileOpenofficeDraw from '@ulaval/modul-components/dist/assets/icons/svg/file-openoffice-draw.svg';
import MSvgFileOpenofficeImpress from '@ulaval/modul-components/dist/assets/icons/svg/file-openoffice-impress.svg';
import MSvgFileOpenofficeMath from '@ulaval/modul-components/dist/assets/icons/svg/file-openoffice-math.svg';
import MSvgFileOpenofficeWritter from '@ulaval/modul-components/dist/assets/icons/svg/file-openoffice-writter.svg';
import MSvgFilePdf from '@ulaval/modul-components/dist/assets/icons/svg/file-pdf.svg';
import MSvgFilePowerpoint from '@ulaval/modul-components/dist/assets/icons/svg/file-powerpoint.svg';
import MSvgFileQuicktime from '@ulaval/modul-components/dist/assets/icons/svg/file-quicktime.svg';
import MSvgFileRealplayer from '@ulaval/modul-components/dist/assets/icons/svg/file-realplayer.svg';
import MSvgFileText from '@ulaval/modul-components/dist/assets/icons/svg/file-text.svg';
import MSvgFileVideo from '@ulaval/modul-components/dist/assets/icons/svg/file-video.svg';
import MSvgFileVisio from '@ulaval/modul-components/dist/assets/icons/svg/file-visio.svg';
import MSvgFileWord from '@ulaval/modul-components/dist/assets/icons/svg/file-word.svg';
import MSvgFileZip from '@ulaval/modul-components/dist/assets/icons/svg/file-zip.svg';
import MSvgFolderOpen from '@ulaval/modul-components/dist/assets/icons/svg/folder-open.svg';
import MSvgFolder from '@ulaval/modul-components/dist/assets/icons/svg/folder.svg';
import MSvgFullScreen from '@ulaval/modul-components/dist/assets/icons/svg/full-screen.svg';
import MSvgGridFilled from '@ulaval/modul-components/dist/assets/icons/svg/grid-filled.svg';
import MSvgGrid from '@ulaval/modul-components/dist/assets/icons/svg/grid.svg';
import MSvgGrouping from '@ulaval/modul-components/dist/assets/icons/svg/grouping.svg';
import MSvgHeraldry from '@ulaval/modul-components/dist/assets/icons/svg/heraldry.svg';
import MSvgHide from '@ulaval/modul-components/dist/assets/icons/svg/hide.svg';
import MSvgHint from '@ulaval/modul-components/dist/assets/icons/svg/hint.svg';
import MSvgHome from '@ulaval/modul-components/dist/assets/icons/svg/home.svg';
import MSvgImageSlideshow from '@ulaval/modul-components/dist/assets/icons/svg/image-slideshow.svg';
import MSvgImageSquare from '@ulaval/modul-components/dist/assets/icons/svg/image-square.svg';
import MSvgImage1Filled from '@ulaval/modul-components/dist/assets/icons/svg/image1-filled.svg';
import MSvgImage2Filled from '@ulaval/modul-components/dist/assets/icons/svg/image2-filled.svg';
import MSvgInformationFilled from '@ulaval/modul-components/dist/assets/icons/svg/information-filled.svg';
import MSvgInformationWhiteFilled from '@ulaval/modul-components/dist/assets/icons/svg/information-white-filled.svg';
import MSvgInformation from '@ulaval/modul-components/dist/assets/icons/svg/information.svg';
import MSvgLeaveFullScreen from '@ulaval/modul-components/dist/assets/icons/svg/leave-full-screen.svg';
import MSvgListFilled from '@ulaval/modul-components/dist/assets/icons/svg/list-filled.svg';
import MSvgList from '@ulaval/modul-components/dist/assets/icons/svg/list.svg';
import MSvgLock from '@ulaval/modul-components/dist/assets/icons/svg/lock.svg';
import MSvgLogout from '@ulaval/modul-components/dist/assets/icons/svg/logout.svg';
import MSvgMultimediaExternal from '@ulaval/modul-components/dist/assets/icons/svg/multimedia-external.svg';
import MSvgMultimediaFilled from '@ulaval/modul-components/dist/assets/icons/svg/multimedia-filled.svg';
import MSvgMultimedia from '@ulaval/modul-components/dist/assets/icons/svg/multimedia.svg';
import MSvgNotAllowed from '@ulaval/modul-components/dist/assets/icons/svg/not-allowed.svg';
import MSvgNotification from '@ulaval/modul-components/dist/assets/icons/svg/notification.svg';
import MSvgOptions from '@ulaval/modul-components/dist/assets/icons/svg/options.svg';
import MSvgPanel from '@ulaval/modul-components/dist/assets/icons/svg/panel.svg';
import MSvgProfile from '@ulaval/modul-components/dist/assets/icons/svg/profile.svg';
import MSvgRemoveAssignation from '@ulaval/modul-components/dist/assets/icons/svg/remove-assignation.svg';
import MSvgRemoveCircleFilled from '@ulaval/modul-components/dist/assets/icons/svg/remove-circle-filled.svg';
import MSvgRemoveCircle from '@ulaval/modul-components/dist/assets/icons/svg/remove-circle.svg';
import MSvgReplace from '@ulaval/modul-components/dist/assets/icons/svg/replace.svg';
import MSvgRightAnswer from '@ulaval/modul-components/dist/assets/icons/svg/right-answer.svg';
import MSvgRoleAssignation from '@ulaval/modul-components/dist/assets/icons/svg/role-assignation.svg';
import MSvgSearch from '@ulaval/modul-components/dist/assets/icons/svg/search.svg';
import MSvgShow from '@ulaval/modul-components/dist/assets/icons/svg/show.svg';
import MSvgSubtitle from '@ulaval/modul-components/dist/assets/icons/svg/subtitle.svg';
import MSvgText1 from '@ulaval/modul-components/dist/assets/icons/svg/text1.svg';
import MSvgText2 from '@ulaval/modul-components/dist/assets/icons/svg/text2.svg';
import MSvgVideoFilled from '@ulaval/modul-components/dist/assets/icons/svg/video-filled.svg';
import MSvgVideoSlideshow from '@ulaval/modul-components/dist/assets/icons/svg/video-slideshow.svg';
import MSvgVideo from '@ulaval/modul-components/dist/assets/icons/svg/video.svg';
import MSvgWarningFilled from '@ulaval/modul-components/dist/assets/icons/svg/warning-filled.svg';
import MSvgWarningWhiteFilled from '@ulaval/modul-components/dist/assets/icons/svg/warning-white-filled.svg';
import MSvgWarning from '@ulaval/modul-components/dist/assets/icons/svg/warning.svg';
import MSvgWrongAnswer from '@ulaval/modul-components/dist/assets/icons/svg/wrong-answer.svg';
import { ModulIconName } from '@ulaval/modul-components/dist/utils/modul-icons/modul-icons';
import { SvgSpriteService } from '@ulaval/modul-components/dist/utils/svg/svg-sprite';
import Vue from 'vue';




export const importAllSvg: () => void = () => {
    const svg: SvgSpriteService = Vue.prototype.$svgSprite;
    svg.addSvg(ModulIconName.AddCircleFilled, MSvgAddCircleFilled);
    svg.addSvg(ModulIconName.AddCircle, MSvgAddCircle);
    svg.addSvg(ModulIconName.ArrowDown, MSvgArrowDown);
    svg.addSvg(ModulIconName.ArrowLeft, MSvgArrowLeft);
    svg.addSvg(ModulIconName.ArrowRight, MSvgArrowRight);
    svg.addSvg(ModulIconName.ArrowUp, MSvgArrowUp);
    svg.addSvg(ModulIconName.ArrowButtonDown, MSvgArrowButtonDown);
    svg.addSvg(ModulIconName.ArrowButtonRight, MSvgArrowButtonRight);
    svg.addSvg(ModulIconName.ArrowButtonLeft, MSvgArrowButtonLeft);
    svg.addSvg(ModulIconName.ArrowButtonUp, MSvgArrowButtonUp);
    svg.addSvg(ModulIconName.ArrowHeadCircleRight, MSvgArrowHeadCircleRight);
    svg.addSvg(ModulIconName.ArrowHeadFilledDown, MSvgArrowHeadFilledDown);
    svg.addSvg(ModulIconName.ArrowHeadFilledLeft, MSvgArrowHeadFilledLeft);
    svg.addSvg(ModulIconName.ArrowHeadFilledRight, MSvgArrowHeadFilledRight);
    svg.addSvg(ModulIconName.ArrowHeadFilledUp, MSvgArrowHeadFilledUp);
    svg.addSvg(ModulIconName.ArrowReturn, MSvgArrowReturn);
    svg.addSvg(ModulIconName.ArrowThinDown, MSvgArrowThinDown);
    svg.addSvg(ModulIconName.ArrowThinLeft, MSvgArrowThinLeft);
    svg.addSvg(ModulIconName.ArrowThinRight, MSvgArrowThinRight);
    svg.addSvg(ModulIconName.ArrowThinUp, MSvgArrowThinUp);
    svg.addSvg(ModulIconName.Attachment, MSvgAttachment);
    svg.addSvg(ModulIconName.Avatar, MSvgAvatar);
    svg.addSvg(ModulIconName.Calendar, MSvgCalendar);
    svg.addSvg(ModulIconName.ChevronDown, MSvgChevronDown);
    svg.addSvg(ModulIconName.ChevronLeft, MSvgChevronLeft);
    svg.addSvg(ModulIconName.ChevronRight, MSvgChevronRight);
    svg.addSvg(ModulIconName.ChevronUp, MSvgChevronUp);
    svg.addSvg(ModulIconName.Clock, MSvgClock);
    svg.addSvg(ModulIconName.CloseClear, MSvgCloseClear);
    svg.addSvg(ModulIconName.CodeTag, MSvgCodeTag);
    svg.addSvg(ModulIconName.CompletedFilled, MSvgCompletedFilled);
    svg.addSvg(ModulIconName.ConfirmationWhiteFilled, MSvgConfirmationWhiteFilled);
    svg.addSvg(ModulIconName.Confirmation, MSvgConfirmation);
    svg.addSvg(ModulIconName.Delete, MSvgDelete);
    svg.addSvg(ModulIconName.Download, MSvgDownload);
    svg.addSvg(ModulIconName.DragVertical, MSvgDragVertical);
    svg.addSvg(ModulIconName.Dragndrop, MSvgDragndrop);
    svg.addSvg(ModulIconName.Edit, MSvgEdit);
    svg.addSvg(ModulIconName.EmailSend, MSvgEmailSend);
    svg.addSvg(ModulIconName.Email, MSvgEmail);
    svg.addSvg(ModulIconName.ErrorFilled, MSvgErrorFilled);
    svg.addSvg(ModulIconName.ErrorWhiteFilled, MSvgErrorWhiteFilled);
    svg.addSvg(ModulIconName.Error, MSvgError);
    svg.addSvg(ModulIconName.Evaluation, MSvgEvaluation);
    svg.addSvg(ModulIconName.FileAccess, MSvgFileAccess);
    svg.addSvg(ModulIconName.FileAudio, MSvgFileAudio);
    svg.addSvg(ModulIconName.FileCode, MSvgFileCode);
    svg.addSvg(ModulIconName.FileDefault, MSvgFileDefault);
    svg.addSvg(ModulIconName.FileDwg, MSvgFileDwg);
    svg.addSvg(ModulIconName.FileExcel, MSvgFileExcel);
    svg.addSvg(ModulIconName.FileFlash, MSvgFileFlash);
    svg.addSvg(ModulIconName.FileImage, MSvgFileImage);
    svg.addSvg(ModulIconName.FileMarkup, MSvgFileMarkup);
    svg.addSvg(ModulIconName.FileMediaplayer, MSvgFileMediaplayer);
    svg.addSvg(ModulIconName.FileOpenofficeBase, MSvgFileOpenofficeBase);
    svg.addSvg(ModulIconName.FileOpenofficeCalc, MSvgFileOpenofficeCalc);
    svg.addSvg(ModulIconName.FileOpenofficeDefault, MSvgFileOpenofficeDefault);
    svg.addSvg(ModulIconName.FileOpenofficeDraw, MSvgFileOpenofficeDraw);
    svg.addSvg(ModulIconName.FileOpenofficeImpress, MSvgFileOpenofficeImpress);
    svg.addSvg(ModulIconName.FileOpenofficeMath, MSvgFileOpenofficeMath);
    svg.addSvg(ModulIconName.FileOpenofficeWritter, MSvgFileOpenofficeWritter);
    svg.addSvg(ModulIconName.FilePdf, MSvgFilePdf);
    svg.addSvg(ModulIconName.FilePowerpoint, MSvgFilePowerpoint);
    svg.addSvg(ModulIconName.FileQuicktime, MSvgFileQuicktime);
    svg.addSvg(ModulIconName.FileRealplayer, MSvgFileRealplayer);
    svg.addSvg(ModulIconName.FileText, MSvgFileText);
    svg.addSvg(ModulIconName.FileVideo, MSvgFileVideo);
    svg.addSvg(ModulIconName.FileVisio, MSvgFileVisio);
    svg.addSvg(ModulIconName.FileWord, MSvgFileWord);
    svg.addSvg(ModulIconName.FileZip, MSvgFileZip);
    svg.addSvg(ModulIconName.FolderOpen, MSvgFolderOpen);
    svg.addSvg(ModulIconName.Folder, MSvgFolder);
    svg.addSvg(ModulIconName.FullScreen, MSvgFullScreen);
    svg.addSvg(ModulIconName.GridFilled, MSvgGridFilled);
    svg.addSvg(ModulIconName.Grid, MSvgGrid);
    svg.addSvg(ModulIconName.Grouping, MSvgGrouping);
    svg.addSvg(ModulIconName.Heraldry, MSvgHeraldry);
    svg.addSvg(ModulIconName.Hide, MSvgHide);
    svg.addSvg(ModulIconName.Hint, MSvgHint);
    svg.addSvg(ModulIconName.Home, MSvgHome);
    svg.addSvg(ModulIconName.ImageSlideshow, MSvgImageSlideshow);
    svg.addSvg(ModulIconName.ImageSquare, MSvgImageSquare);
    svg.addSvg(ModulIconName.Image1Filled, MSvgImage1Filled);
    svg.addSvg(ModulIconName.Image2Filled, MSvgImage2Filled);
    svg.addSvg(ModulIconName.InformationFilled, MSvgInformationFilled);
    svg.addSvg(ModulIconName.InformationWhiteFilled, MSvgInformationWhiteFilled);
    svg.addSvg(ModulIconName.Information, MSvgInformation);
    svg.addSvg(ModulIconName.LeaveFullScreen, MSvgLeaveFullScreen);
    svg.addSvg(ModulIconName.ListFilled, MSvgListFilled);
    svg.addSvg(ModulIconName.List, MSvgList);
    svg.addSvg(ModulIconName.Lock, MSvgLock);
    svg.addSvg(ModulIconName.Logout, MSvgLogout);
    svg.addSvg(ModulIconName.MultimediaExternal, MSvgMultimediaExternal);
    svg.addSvg(ModulIconName.MultimediaFilled, MSvgMultimediaFilled);
    svg.addSvg(ModulIconName.Multimedia, MSvgMultimedia);
    svg.addSvg(ModulIconName.NotAllowed, MSvgNotAllowed);
    svg.addSvg(ModulIconName.Notification, MSvgNotification);
    svg.addSvg(ModulIconName.Options, MSvgOptions);
    svg.addSvg(ModulIconName.Panel, MSvgPanel);
    svg.addSvg(ModulIconName.Profile, MSvgProfile);
    svg.addSvg(ModulIconName.RemoveAssignation, MSvgRemoveAssignation);
    svg.addSvg(ModulIconName.RemoveCircleFilled, MSvgRemoveCircleFilled);
    svg.addSvg(ModulIconName.RemoveCircle, MSvgRemoveCircle);
    svg.addSvg(ModulIconName.Replace, MSvgReplace);
    svg.addSvg(ModulIconName.RightAnswer, MSvgRightAnswer);
    svg.addSvg(ModulIconName.RoleAssignation, MSvgRoleAssignation);
    svg.addSvg(ModulIconName.Search, MSvgSearch);
    svg.addSvg(ModulIconName.Show, MSvgShow);
    svg.addSvg(ModulIconName.Subtitle, MSvgSubtitle);
    svg.addSvg(ModulIconName.Text1, MSvgText1);
    svg.addSvg(ModulIconName.Text2, MSvgText2);
    svg.addSvg(ModulIconName.VideoFilled, MSvgVideoFilled);
    svg.addSvg(ModulIconName.VideoSlideshow, MSvgVideoSlideshow);
    svg.addSvg(ModulIconName.Video, MSvgVideo);
    svg.addSvg(ModulIconName.WarningFilled, MSvgWarningFilled);
    svg.addSvg(ModulIconName.WarningWhiteFilled, MSvgWarningWhiteFilled);
    svg.addSvg(ModulIconName.Warning, MSvgWarning);
    svg.addSvg(ModulIconName.WrongAnswer, MSvgWrongAnswer);
};
