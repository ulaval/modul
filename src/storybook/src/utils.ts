export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';
export const modulWebsiteHierarchyRootSeparator: string = 'modul-website|';
