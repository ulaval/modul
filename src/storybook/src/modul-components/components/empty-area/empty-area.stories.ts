import { actions } from '@storybook/addon-actions';
import { select, text } from '@storybook/addon-knobs';
import { EMPTY_AREA_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MEmptyAreaButtonType } from '@ulaval/modul-components/dist/components/empty-area/empty-area';
import { Enums } from '@ulaval/modul-components/dist/utils/enums/enums';
import { ModulIconName } from '@ulaval/modul-components/dist/utils/modul-icons/modul-icons';
import { Component } from 'vue';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';
import { importAllSvg } from '../svg/svg-importation';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${EMPTY_AREA_NAME}`,
    parameters: { fileName: __filename }
};

export const Sandbox = () => ({
    methods: actions('emitButtonClick'),
    beforeCreate() {
        importAllSvg();
    },
    props: {
        title: {
            default: text('Prop title', 'No content available')
        },
        subtitle: {
            default: text('Prop subtitle', '')
        },
        buttonText: {
            default: text('Prop button-text', 'Add')
        },
        buttonType: {
            default: select(
                'Prop button-type',
                Enums.toValueArray(MEmptyAreaButtonType),
                MEmptyAreaButtonType.AddButton
            )
        },
        svgName: {
            default: select<ModulIconName>(
                'Prop name',
                Enums.toValueArray(ModulIconName) as ModulIconName[],
                ModulIconName.Folder
            )
        }
    },
    template: `<${EMPTY_AREA_NAME}
        :title="title"
        :subtitle="subtitle"
        :button-text="buttonText"
        :button-type="buttonType"
        :svg-name="svgName"
        @button-click="emitButtonClick"
    />`
});

export const PropSvgName: () => Component = (): Component => ({
    beforeCreate() {
        importAllSvg();
    },
    props: {
        svgName: {
            default: select<ModulIconName>(
                'Prop name',
                Enums.toValueArray(ModulIconName) as ModulIconName[],
                ModulIconName.Calendar
            )
        }
    },
    template: `<${EMPTY_AREA_NAME}
        :svg-name="svgName"
    />`
});

export const PropTitle: () => Component = (): Component => ({
    props: {
        title: {
            default: text('Prop title', 'Prop title')
        }
    },
    template: `<${EMPTY_AREA_NAME}
        :title="title"
    />`
});

export const PropSubtitle: () => Component = (): Component => ({
    props: {
        subtitle: {
            default: text('Prop subtitle', 'Prop subtitle')
        }
    },
    template: `<${EMPTY_AREA_NAME}
        :subtitle="subtitle"
    />`
});

export const PropButtonText: () => Component = (): Component => ({
    beforeCreate() {
        importAllSvg();
    },
    props: {
        buttonText: {
            default: text('Prop button-text', 'Prop button-text')
        }
    },
    methods: actions('emitButtonClick'),
    template: `<${EMPTY_AREA_NAME}
        :button-text="buttonText"
        @button-click="emitButtonClick"
    />`
});

export const PropButtonType: () => Component = (): Component => ({
    methods: actions('emitButtonClick'),
    template: `<div>
        <${EMPTY_AREA_NAME}
            button-text="Prop button-type = ${MEmptyAreaButtonType.Button}"
            button-type="${MEmptyAreaButtonType.Button}"
            @button-click="emitButtonClick"
        />
        <${EMPTY_AREA_NAME}
            class="m-u--margin-top--l"
            button-text="Prop button-type = ${MEmptyAreaButtonType.AddButton}"
            button-type="${MEmptyAreaButtonType.AddButton}"
            @button-click="emitButtonClick"
        />
    </div>`
});
