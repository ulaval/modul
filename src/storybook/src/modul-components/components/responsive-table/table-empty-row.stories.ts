import { actions } from '@storybook/addon-actions';
import { RESPONSIVE_TABLE_NAME, TABLE_EMPTY_ROW_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MEmptyAreaButtonType } from '@ulaval/modul-components/dist/components/empty-area/empty-area';
import { ModulIconName } from '@ulaval/modul-components/dist/utils/modul-icons/modul-icons';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';
import { importAllSvg } from '../svg/svg-importation';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${RESPONSIVE_TABLE_NAME}/${TABLE_EMPTY_ROW_NAME}`,
    parameters: { fileName: __filename }
};

export const Default = () => ({
    data: () => ({
        emptyArea: {
            title: 'Tilte',
            subtitle: 'Subtitle',
            svgName: ModulIconName.Folder,
            buttonText: 'Add content',
            buttonType: MEmptyAreaButtonType.Button
        }
    }),
    methods: actions('emitEmptyButtonClick'),
    beforeCreate() {
        importAllSvg();
    },
    template: `<${TABLE_EMPTY_ROW_NAME}
        :empty-area="emptyArea"
        :table-component-width="'600px'"
        :horizontal-scroll-offset="10"
        :nb-columns="3"
        @empty-button-click="emitEmptyButtonClick"
    />`
});

export const PropWaiting = () => ({
    data: () => ({
        emptyArea: {
            title: 'Text',
            svgName: ModulIconName.Folder,
            buttonText: 'Add content',
            buttonType: MEmptyAreaButtonType.Button
        }
    }),
    methods: actions('emitEmptyButtonClick'),
    beforeCreate() {
        importAllSvg();
    },
    template: `<${TABLE_EMPTY_ROW_NAME}
        :empty-area="emptyArea"
        :table-component-width="'600px'"
        :horizontal-scroll-offset="10"
        :waiting="true"
        @empty-button-click="emitEmptyButtonClick"
    />`
});

export const PropInsideOfTableGroup = () => ({
    data: () => ({
        emptyArea: {
            title: 'Prop inside-of-table-group = true',
            svgName: ModulIconName.Folder
        }
    }),
    beforeCreate() {
        importAllSvg();
    },
    template: `<${TABLE_EMPTY_ROW_NAME}
        :empty-area="emptyArea"
        :table-component-width="'600px'"
        :horizontal-scroll-offset="10"
        :inside-of-table-group="true"
    />`
});
