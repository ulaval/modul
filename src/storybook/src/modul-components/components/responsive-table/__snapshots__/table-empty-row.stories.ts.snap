// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-responsive-table/m-table-empty-row Default 1`] = `
<tr
  class="m-table-empty-row"
>
  <td
    colspan="3"
  >
    <div
      class="m-empty-area"
      style="left: 10px; width: 600px;"
    >
      <button
        class="m-button m--is-skin-secondary"
        type="button"
      >
        <!---->
         
        <!---->
         
        <span
          class="m-button__text"
        >
          
            Add content
         
          <!---->
        </span>
         
        <!---->
         
        <!---->
      </button>
       
      <div
        class="m-empty-area__icon-area"
      >
        <!---->
         
        <p
          class="m-empty-area__title"
        >
          <strong>
            Tilte
          </strong>
        </p>
         
        <p
          class="m-empty-area__subtitle"
        >
          
            Subtitle
        
        </p>
      </div>
    </div>
  </td>
</tr>
`;

exports[`Storyshots modul-components|m-responsive-table/m-table-empty-row Waiting 1`] = `
<tr
  class="m-table-empty-row"
>
  <td>
    <div
      class="m-empty-area"
      style="left: 10px; width: 600px;"
    >
      <!---->
       
      <div
        class="m-empty-area__icon-area"
      >
        <!---->
         
        <p
          class="m-empty-area__title"
        >
          <strong>
            Chargement en cours
          </strong>
        </p>
         
        <!---->
      </div>
    </div>
  </td>
</tr>
`;
